### ISSUE TITLE
Memory usage in covergeBed 2.24

### ISSUE THREAD

**Issue by dariober** [arq5x/bedtools2#282]
Using coverageBed v2.24.0 to count reads in bam file overlapping features in bed file. It appears the bam is loaded in memory thus requiring a huge amount of memory. Here's the log from LSF bsub:

```
# LSBATCH: User input
coverageBed -b reh011_wtHEKfUlib16.150731.bam -a reh011_wtHEKfUlib16.150731_peaks.narrowPeak
------------------------------------------------------------

TERM_MEMLIMIT: job killed after reaching LSF memory usage limit.
Exited with exit code 130.

Resource usage summary:

    CPU time   :     30.45 sec.
    Max Memory :      7519 MB
    Max Swap   :      7671 MB

    Max Processes  :         4
    Max Threads    :         5
```
Input bam is 3.1GB, and the bed file has just 39 features of few hundred bases each.

Thanks!
Dario

**Comment by arq5x** [comment]
Yes, you would need to use the `-sorted` option to avoid loading the BAM file into memory.

**Comment by dariober** [comment]
Hi Aaron- Thanks. So there is no plan to add an option to keep the bam file out of memory, just like in v2.23 and earlier?

**Comment by arq5x** [comment]
I am not sure to what you are referring.

**Comment by dariober** [comment]
Sorry- I was asking if in v2.24+ it will be possible to load the interval file in memory and scan the bam file line by line. 
If I'm not mistaken in previous versions of coveragebed the b-file was loaded in memory and the a-file read line by line, right? So if you did ```coverageBed -abam reads.bam -b intervals.bed``` the bam file was read line by line without too much memory usage (and without the need of sorting it). This was great for counting reads in features.
The current -sorted option is great, but the annoying thing is that bam files are sometime sorted in weird chromosome orders (like chrM at the beginning or at the end) and I'd rather avoid re-sorting them (although in the long run is probably a good thing anyway?). Does it make sense...?
By the way, I see there was some discussion about it here http://bedtools.readthedocs.org/en/latest/content/tools/coverage.html from Matt Shirley.

Dario

**Comment by sri-jumpcode** [comment]
I ran into the same problem. The messages here helped me understand the issues with bedtools well. I wanted to thank everyone for the comments here. The way I resolved it was:

1. Indexed the BAM file using samtools
2. Filtered reads from BAM file for my region of interest (samtools view)
3. Ran bedtools coverage -a [bed file] -b [filtered BAM] -bed -d > [output]

Surprisingly, this was faster than running bedtools even for smaller files.
